services/identity-service/src/did/did.service.ts

```diff
@@ -0,0 +1,174 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import * as ION from '@decentralized-identity/ion-tools';
+import { PrismaService } from 'src/prisma.service';
+import { Identity, Prisma } from '@prisma/client';
+import { DIDDocument } from 'did-resolver';
+import { domainToASCII } from 'url';
+import { uuid } from 'uuidv4';
+import { GenerateDidDTO } from './dtos/GenerateDid.dto';
+import { VaultService } from './vault.service';
+
+@Injectable()
+export class DidService {
+  constructor(private prisma: PrismaService, private vault: VaultService) {}
+
+  async generateDID(doc: GenerateDidDTO): Promise<DIDDocument> {
+    // Create private/public key pair
+    const authnKeys = await ION.generateKeyPair('Ed25519');
+    console.log(authnKeys);
+
+    // Create a UUID for the DID using uuidv4
+    const didUri = 'did:ulp:' + uuid();
```

Comment:
remove ULP keyword and make it configurable

Reply:
done refer: https://github.com/Sunbird-RC/sunbird-rc-core/pull/218/commits/6af1cd1051d87dacc503de18f58ca85203c99ac4